app/frontend/components/AccountSummaryModal.tsx

```diff
@@ -0,0 +1,93 @@
+import React, { ComponentProps, useEffect, useState, useCallback } from 'react'
+import { nsEventName, LoadAccountEventDetail } from '@/utils'
+
+const AccountSummaryModal: React.FC<ComponentProps<'div'>> = ({ children, id, ...props }) => {
+  const [accountLoader] = useState<AbortController>(() => new AbortController())
+  const modalId = id || 'account--summary-modal'
+
+  /**
+   * React query functions: https://tanstack.com/query/latest/docs/framework/react/guides/query-functions
+   * - variables: https://tanstack.com/query/latest/docs/framework/react/guides/query-functions#query-function-variables   * React query options: https://tanstack.com/query/latest/docs/framework/react/guides/query-options
+   * Using react query with fetch: https://tanstack.com/query/latest/docs/framework/react/guides/query-functions#usage-with-fetch-and-other-clients-that-do-not-throw-by-default
+   */
+  const listenForAccountLoadEvents = useCallback(() => {
```

Comment:
This is a whole thing for sending data to the client! 😅 Can it be a `@/util`? Also, right now it will require a data request. Is there a way to use sockets or something to avoid the round trip via HTTP, react query + fetch etc.?